### ISSUE TITLE
User story: unnecessary use of mutable smart pointer

### ISSUE THREAD

**Issue by alice-i-cecile** [rust-linting/design#25]
### Lint explanation

Smart pointers often come in two forms. In bevy's case, the most common are typically `Res` and `ResMut`. 

Users should only use this when necessary (i.e. when they actually intend to mutate the data), and should request the less powerful (more easily parallelized) immutable variant when they can.

As a result, using requesting a `ResMut` without actually making it mutable is almost always a mistake.

### Example code

```rust
fn my_system(time: ResMut<Time>) {}
```

Should be

```rust
fn my_system(time: Res<Time>) {}
```

### Notes

False positives may exist, e.g. with interior mutability, so advanced users need a way to locally allow this.
